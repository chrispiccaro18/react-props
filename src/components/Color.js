import React from 'react';
import PropTypes from 'prop-types';

function Color({ name, rgb }) {
  const hex = '#' + rgb.red.toString(16).toUpperCase() + rgb.green.toString(16).toUpperCase() + rgb.blue.toString(16).toUpperCase();
  
  const colorPreview = {
    backgroundColor: hex,
    display: 'inline-block',
    width: '0.8em',
    height: '0.8em'
  };

  return (<dl>
    <dt>Name</dt>
    <dd>
      {name ? name : hex}
      <div style={colorPreview}></div>
    </dd>

    <dt>Hex</dt>
    <dd>{hex}</dd>

    <dt>RGB</dt>
    <dd>Red - {rgb.red}</dd>
    <dd>Green - {rgb.green}</dd>
    <dd>Blue - {rgb.blue}</dd>
  </dl>
  );
}

Color.propTypes = {
  name: PropTypes.string,
  rgb: PropTypes.object.isRequired
};

export default Color;
