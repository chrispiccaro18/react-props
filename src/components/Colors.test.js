import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color component', () => {
  it('renders a Color', () => {
    const wrapper = shallow(<Color name='sky blue' rgb={{ red: 135, green: 206, blue: 235 }}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
