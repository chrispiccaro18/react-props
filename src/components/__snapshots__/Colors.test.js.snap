// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color component renders a Color 1`] = `
<dl>
  <dt>
    Name
  </dt>
  <dd>
    sky blue
    <div
      style={
        Object {
          "backgroundColor": "#87CEEB",
          "display": "inline-block",
          "height": "0.8em",
          "width": "0.8em",
        }
      }
    />
  </dd>
  <dt>
    Hex
  </dt>
  <dd>
    #87CEEB
  </dd>
  <dt>
    RGB
  </dt>
  <dd>
    Red - 
    135
  </dd>
  <dd>
    Green - 
    206
  </dd>
  <dd>
    Blue - 
    235
  </dd>
</dl>
`;
